File: Linear/Static/Array/03_Deleting/Given_Position/Custom/Deleting.js
import readline from 'readline'

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

function deleteAtGivenPosition(arr, pos) {    
    // Delete the element at the given position.
    for (let i = pos; i < arr.length; i++) {
        arr[i - 1] = arr[i]
    }

    arr.length--

    return arr
}

let arr = [10, 20, 30, 40]

console.log("Array before deletion")
console.log(arr.join(" "))

rl.question("\nEnter a position: ", (ans) => {
    const pos = parseInt(ans)

    deleteAtGivenPosition(arr, pos)

    console.log("Array after deletion")
    console.log(arr.join(" "))
    
    rl.close()
})

File: Linear/Static/Array/03_Deleting/Given_Position/Custom/Deleting.py
def deleteAtGivenPosition(arr, pos):
    # Delete the element at the given position.
    for i in range(pos, len(arr)):
        arr[i - 1] = arr[i]

    arr.pop()

    return arr

arr = [10, 20, 30, 40]

print("Array before deletion")
for i in range(len(arr)):
    print(arr[i], end=" ")

pos = int(input("\nEnter a position: "))

deleteAtGivenPosition(arr, pos)

print("\nArray after deletion")
for i in range(len(arr)):
    print(arr[i], end=" ")